### ISSUE TITLE
Bad API for nullable Either Types

### ISSUE THREAD

**Issue by cdietrich** [eclipse/lsp4j#262]
given a lsp4j api like `CompletionItem` with `documentation?: string | MarkupContent`

we get following api

```
  /**
   * A human-readable string that represents a doc-comment.
   */
  public void setDocumentation(final Either<String, MarkupContent> documentation) {
    this.documentation = documentation;
  }
  
  public void setDocumentation(final String documentation) {
    this.documentation = Either.forLeft(documentation);
  }
  
  public void setDocumentation(final MarkupContent documentation) {
    this.documentation = Either.forRight(documentation);
  }
```

if this is now called with a `null` `String` or a `null` `MarkupContent`
e.g. as it might happen here
https://github.com/eclipse/xtext-core/blob/9e46f302bb7bfd68b61f1a05f1c9bb5d830b7fa1/org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/contentassist/ContentAssistService.xtend#L95

this will lead to a double null either type.
since the Either types can change even with stable `setDocumentation` i am not sure if the Client side is the right place for that or if the problem should be treated in lsp4j api e.g.

```
 public void setDocumentation(final String documentation) {
    if (documentation == null)
        this.documentation = null
    else
    this.documentation = Either.forLeft(documentation);
  }
```

this issue was originally reported as

https://github.com/eclipse/xtext-eclipse/issues/834

**Comment by cdietrich** [comment]
please let me know if you need any other information

**Comment by dhuebner** [comment]
Fixed with #269
